### PR TITLE
Add Python documentation [Sphinx-Hatch]

### PR DESCRIPTION
**Pull request by RaulTrombin**
This PR add the sphinx source folder to autogenerate the documentation for Python module.
This current project uses the napoleon module, [sphinx.ext.napoleon](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#module-sphinx.ext.napoleon) – Support for NumPy and Google style docstrings[¶](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#module-sphinx.ext.napoleon)
It uses the upgrades made on cpy-binder 3.0, which now support comments.
This PR also create an action for page deployment.

To test it locally:
 `pip install -r docs/requirements.txt && maturin build && env pip install --force-reinstall target/wheels/*.whl && cd docs && make clean html`

Deployments:
You can test it on a raspberry with navigator:
pip install br-navigator
https://pypi.org/project/br-navigator/
https://raultrombin.github.io/navigator-lib/
ps.: here I'm using the br-navigator name for a preview.

**Previews:**
image
image

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+templates_path = ['_templates']
+exclude_patterns = []
+
+# -- Options for HTML output -------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output
+
+html_theme = 'sphinx_rtd_theme'
+html_static_path = ['_static']
+html_logo = "https://upload.wikimedia.org/wikipedia/commons/1/12/Bluerobotics-logo.svg"
```
Comment (patrickelectric) [inline docs/source/conf.py:38]: Can we use this https://github.com/sphinx-toolbox/sphinx-pyproject to move mostly this configuration to pyproject ?

File: docs/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+maturin
+sphinx
+sphinx-rtd-theme
+sphinxcontrib-napoleon
```
Comment (patrickelectric) [inline docs/requirements.txt:4]: This could be moved to pyproject as well

File: docs/source/conf.py (hunk 1 above)
Comment (RaulTrombin) [reply docs/source/conf.py]: we can, no stronger feelings on this, we still need a conf.py, just the content is moved to pyproject.toml file.

File: docs/requirements.txt (hunk 2 above)
Comment (RaulTrombin) [reply docs/requirements.txt:4]: yes it becames unnecessary if we use poetry to setup the enviroment. please check new commits.

File: docs/Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
[the remark this answers is not in the document]
Comment (RaulTrombin) [reply docs/Makefile:20]: not have sucess on this yet.
but this is currently working like this:
```
 6709  poetry install
 6711  poetry shell 
 6713  maturin build && env pip install --force-reinstall target/wheels/*.whl && cd docs && make clean html
```

**Comment by RaulTrombin** [comment]
removed makefile and requirements.txt.

Now we have hatch

hatch run dev:build-doc

**Review by patrickelectric — CHANGES_REQUESTED** [review]
@RaulTrombin CI is failing, needs rebase

**Comment by RaulTrombin** [comment]
> @RaulTrombin CI is failing, needs rebase

rebased

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**